src/main/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/azuredevops/model/CommentThread.java

```diff
@@ -0,0 +1,107 @@
+package com.github.mc1arke.sonarqube.plugin.ce.pullrequest.azuredevops.model;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.github.mc1arke.sonarqube.plugin.ce.pullrequest.azuredevops.model.enums.CommentThreadStatus;
+import org.sonar.db.protobuf.DbIssues;
+
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.List;
+
+/**
+ * Represents a comment thread of a pull request. A thread contains meta data about the file
+ * it was left on along with one or more comments (an initial comment and the subsequent replies).
+ */
+public class CommentThread implements Serializable {
+
+    private CommentThreadStatus status;
+    private List<Comment> comments;
+    private CommentThreadContext threadContext;
+    private int id;
+    private Date publishedDate;
+    private Date lastUpdatedDate;
+    private HashMap<String, IdentityRef> identities;
+    @JsonProperty("isDeleted")
+    private boolean deleted;
+    @JsonProperty("_links")
+    private ReferenceLinks links;
+
+    public CommentThread(){};
```

Comment:
Could you confirm what the default constructors are being used for in your change please? They prevent fields being made final which is a convention being used to enforce thread safety

Reply:
Updated to include JsonCreator constructor and made all fields final.